extension/src/Registry.ts

```diff
@@ -145,29 +157,32 @@ export class Registry {
      */
     public async getPackages(token?: CancellationToken): Promise<Package[]> {
         const packages: Package[] = [];
-        const config = getConfig();
 
         for await (const result of this.findMatchingPackages(this.query, token)) {
             if (token?.isCancellationRequested) {
                 break;
             }
 
             try {
-                let channel = config.get<any>('channels')?.[result.name] ?? 'latest';
-
-                // Ensure requested channel is valid, otherwise fallback to latest
-                if (channel !== 'latest') {
-                    let meta: any = await this.getPackageMetadata(result.name);
-                    if (!(channel in meta['dist-tags']) && !(channel in meta['versions'])) {
-                        channel = 'latest';
-                    }
-                }
-
-                const manifest = await this.getPackageVersionMetadata(result.name, channel);
-                packages.push(new Package(this, manifest, channel));
+                const pkg = await this.getPackageVersionMetadata(result.name);
+                packages.push(pkg);
             } catch (ex) {
                 if (ex instanceof NotAnExtensionError) {
                     // Package is not an extension. Ignore.
+                } else if (ex instanceof VersionMissingError) {
+                    // Requested package version does not exist
+                    const openSettingsJson = localize('open.settings.json', 'Open settings.json');
+                    const settingsJsonLink = `[${openSettingsJson}](command:workbench.action.openSettingsJson)`;
+
+                    // TODO: Add a quick link to reset to 'latest' via command
+                    window.showErrorMessage(
+                        localize(
+                            'invalid.channel',
+                            '{0} Your "privateExtensions.channels" setting may be invalid.  {1} to fix.',
```

Comment:
Minor: Don't need double spaces after periods. (The rendering engine probably collapses these anyways though.)